addons/graphql/src/shared.ts

```diff
@@ -0,0 +1,22 @@
+import { fetch } from 'global';
+
+import { FetcherParams } from './preview';
+
+const FETCH_OPTIONS = {
+  method: 'post',
+  headers: { 'Content-Type': 'application/json' },
+};
+
+export const getDefaultFetcher = (url: string) => {
+  return (params: FetcherParams) => {
+    const body = JSON.stringify(params);
+    const options = Object.assign({ body }, FETCH_OPTIONS);
+    return fetch(url, options).then((res: any) => res.json());
+  };
+};
+
+export const reIndentQuery = (query: string) => {
+  const lines = query.split('\n');
+  const spaces = lines[lines.length - 1].length - 1;
+  return lines.map((l: string, i: number) => (i === 0 ? l : l.slice(spaces))).join('\n');
```

Comment:
I think types of `l` and `i` will be automatically inferred by TS as soon as you typed `query`, so maybe they are useless

Reply:
Yeap, It is inferred. I should be cautious about next time.